spyglass/src/main/java/com/linkedin/android/spyglass/ui/MentionsEditText.java

```diff
@@ -206,19 +219,37 @@ public QueryToken getQueryTokenIfValid() {
      */
     @Override
     public boolean onTouchEvent(@NonNull MotionEvent event) {
-        // If user tapped a span
-        MentionSpan touchedSpan = getTouchedSpan(event);
-        if (touchedSpan != null) {
-            if (event.getAction() == MotionEvent.ACTION_UP) {
+        final MentionSpan touchedSpan = getTouchedSpan(event);
+        boolean superResult = super.onTouchEvent(event);
+        if (event.getAction() == MotionEvent.ACTION_UP) {
+            // Don't call the onclick on mention if MotionEvent.ACTION_UP is for long click action,
+            if (!isLongPressed && touchedSpan != null) {
                 // Manually click span and show soft keyboard
-                touchedSpan.onClick(this);
+                touchedSpan.onClick(MentionsEditText.this);
                 Context context = getContext();
                 if (context != null) {
                     InputMethodManager imm = (InputMethodManager) context.getSystemService(Context.INPUT_METHOD_SERVICE);
-                    imm.showSoftInput(this, 0);
+                    imm.showSoftInput(MentionsEditText.this, 0);
                 }
+                return true;
             }
-            return true;
+        } else if (event.getAction() == MotionEvent.ACTION_DOWN) {
+            isLongPressed = false;
+            if (touchedSpan != null) {
```

Comment:
I don't think we should allow long presses if the user has set the long clickable view flag to false. By default, I expect most developers to want cut/copy/paste support, but I also think we should allow disabling this functionality if desired.